craft_providers/lxd/remotes.py

```diff
@@ -15,50 +16,165 @@
+                # let the original exception fly)
+                if self.remote_name in lxc.remote_list():
+                    logger.debug(
+                        "Remote %r is present on second check, ignoring exception %r.",
+                        self.remote_name,
+                        exc,
+                    )
+                else:
+                    raise
             else:
-                raise
-        else:
-            logger.debug("Remote %r was successfully added.", BUILDD_REMOTE_NAME)
+                logger.debug("Remote %r was successfully added.", self.remote_name)
+
+
+# XXX: support xenial?
+# mapping from supported bases to actual lxd remote images
+_PROVIDER_BASE_TO_LXD_REMOTE_IMAGE = {
+    BuilddBaseAlias.BIONIC.value: RemoteImage(
+        image_name="core18",
+        remote_name=BUILDD_RELEASES_REMOTE_NAME,
+        remote_address=BUILDD_RELEASES_REMOTE_ADDRESS,
+        remote_protocol=ProtocolType.SIMPLESTREAMS,
+        is_stable=True,
+    ),
+    BuilddBaseAlias.FOCAL.value: RemoteImage(
+        image_name="core20",
+        remote_name=BUILDD_RELEASES_REMOTE_NAME,
+        remote_address=BUILDD_RELEASES_REMOTE_ADDRESS,
+        remote_protocol=ProtocolType.SIMPLESTREAMS,
+        is_stable=True,
+    ),
+    BuilddBaseAlias.JAMMY.value: RemoteImage(
+        image_name="core22",
+        remote_name=BUILDD_RELEASES_REMOTE_NAME,
+        remote_address=BUILDD_RELEASES_REMOTE_ADDRESS,
+        remote_protocol=ProtocolType.SIMPLESTREAMS,
+        is_stable=True,
+    ),
+    BuilddBaseAlias.KINETIC.value: RemoteImage(
+        image_name="kinetic",
+        remote_name=DAILY_REMOTE_NAME,
+        remote_address=DAILY_REMOTE_ADDRESS,
+        remote_protocol=ProtocolType.SIMPLESTREAMS,
+        is_stable=False,
+    ),
+    BuilddBaseAlias.LUNAR.value: RemoteImage(
+        image_name="lunar",
+        remote_name=DAILY_REMOTE_NAME,
+        remote_address=DAILY_REMOTE_ADDRESS,
+        remote_protocol=ProtocolType.SIMPLESTREAMS,
+        is_stable=False,
+    ),
+}
+
+
+def get_remote_image(provider_base: str) -> RemoteImage:
+    """Get a RemoteImage for a particular provider base.
+
+    :param provider_base: string containing the provider base
+
+    :returns: the RemoteImage for the provider base
+    """
+    image = _PROVIDER_BASE_TO_LXD_REMOTE_IMAGE.get(provider_base)
+    if not image:
+        raise LXDError(
+            brief=(
+                "could not find a lxd remote image for the provider base "
+                f"{provider_base!r}"
+            )
+        )
+
+    return image
+
+
+def configure_buildd_image_remote(lxc: LXC = LXC()) -> str:
+    """Configure the default buildd image remote.
+
+    This is a deprecated function to maintain the existing API. It will be
+    removed with the release of craft-providers 2.0.
+
+    :param lxc: LXC client.
+
+    :returns: Name of remote to pass to launcher.
+    """
+    logger.warning(
```

Comment:
I take it we don't use `warnings.warn` and `DeprecationWarning` here because it's filtered out by default?

Reply:
That is what I surmised.  I am open to suggestions for a better approach to deprecations.

Reply:
Updated to DeprecationWarning